service/storage_service.cc

```diff
@@ -538,10 +538,8 @@ class topology_node_mutation_builder
 public:
     topology_node_mutation_builder(topology_mutation_builder&, raft::server_id);
 
-    template<typename T>
-    topology_node_mutation_builder& set(const char* cell, const T& value) {
-        return set(cell, sstring{::format("{}", value)});
-    }
+    topology_node_mutation_builder& set(const char* cell, node_state value);
+    topology_node_mutation_builder& set(const char* cell, topology_request value);
```

Comment:
Can't we restrict the T to be one of two types by a concept?

Reply:
Personally I prefer this version.

Reply:
> Personally I prefer this version.

Just wondering how many duplication it would take for you to prefer concepts. 5, 15, 50? :)